src/lib/encoder.ts

```diff
@@ -352,21 +366,90 @@ const valueTypeEncodingMap = {
     },
   },
   uint256: {
-    encode: (value: string | number) =>
-      abiCoder.encodeParameter('uint256', value),
-    decode: (value: string) => abiCoder.decodeParameter('uint256', value),
+    encode: (value: string | number) => {
+      return abiCoder.encodeParameter('uint256', value);
+    },
+    decode: (value: string) => {
+      if (!isHex(value)) {
+        throw new Error(`Can't convert ${value} to uint256, value is not hex.`);
+      }
+
+      const numberOfBytes = countNumberOfBytes(value);
+
+      if (numberOfBytes > 32) {
+        throw new Error(
+          `Can't convert hex value ${value} to uint256. Too many bytes. ${numberOfBytes} is above the maximal number of bytes 32`,
+        );
+      }
+
+      return BigNumber(value).toNumber();
+    },
   },
   bytes32: {
-    encode: (value) => abiCoder.encodeParameter('bytes32', value),
+    encode: (value: string | number) => {
+      let valueToEncode: string;
+
+      if (typeof value === 'string' && !isHex(value)) {
+        // if we receive a plain string (e.g: "hey!"), convert it to utf8-hex data
+        valueToEncode = toHex(value);
+      } else if (typeof value === 'number') {
+        // if we receive a number as input, convert it to hex
+        valueToEncode = numberToHex(value);
+      } else {
```

Comment:
I think we should add this in the docs. Tell users that if they try to encode a non hex value, we will convert it for them. Maybe you can add an example for each cases:

https://docs.lukso.tech/tools/erc725js/classes/ERC725#encodedata

Here